// ml_service.proto
syntax = "proto3";

service MLService {
    // Load data for training
    rpc LoadData(LoadDataRequest) returns (LoadDataResponse);

    // Train the model
    rpc TrainModel(TrainModelRequest) returns (TrainModelResponse);
}

message LoadDataRequest {
    string dataset_path = 1;
    bytes file_data = 2;
}

message LoadDataResponse {
    bool success = 1;
    string message = 2;
}

message TrainModelRequest {
    int32 epochs = 1;
    float learning_rate = 2;
}

message TrainModelResponse {
    bool success = 1;
    string message = 2;
    float training_accuracy = 3;
}
